test/logger.test.js

```diff
@@ -0,0 +1,229 @@
+      const { __ow_logger: log } = params;
+      log.info('Hello, world.');
+      return {
+        body: 'ok',
+      };
+    }
+    myRootLogger.loggers.set('mylogger', memLogger);
+    const result = await logger.wrap(main, { path: '/foo', SECRET_KEY: 'foobar' }, myRootLogger);
+
+    assert.deepEqual(result, { body: 'ok' });
+
+    assert.deepEqual(memLogger.buf, [{
+      level: 'trace',
+      message: ['before'],
+      params: {
+        path: '/foo',
+      },
+      timestamp: '1970-01-01T00:00:00.000Z',
+    }, {
+      level: 'info',
+      message: ['Hello, world.'],
+      timestamp: '1970-01-01T00:00:00.000Z',
+    }, {
+      level: 'trace',
+      message: ['result'],
+      result: {
+        body: 'ok',
+      },
+      timestamp: '1970-01-01T00:00:00.000Z',
+    }]);
+  });
+
+  it('openhwisk creates coralogix logger if needed', () => {
+    const log = logger.init({
+      CORALOGIX_API_KEY: '1234',
+      CORALOGIX_APPLICATION_NAME: 'logger-test',
+      CORALOGIX_SUBSYSTEM_NAME: 'test-1',
+      CORALOGIX_LOG_LEVEL: 'info',
+    }, myRootLogger);
+
+    const reqs = [];
+    nock('https://api.coralogix.com/api/v1/')
+      .post('/logs')
+      .reply((uri, requestBody) => {
+        reqs.push(requestBody);
+        return [200, 'ok'];
+      });
+
+    log.info({ myId: 42 }, 'Hello, world');
+    assert.equal(reqs.length, 1);
+    assert.equal(reqs[0].applicationName, 'logger-test');
+    assert.equal(reqs[0].subsystemName, 'test-1');
+    assert.equal(reqs[0].privateKey, '1234');
+    assert.equal(reqs[0].logEntries.length, 1);
+
+
+    const logEntry = JSON.parse(reqs[0].logEntries[0].text);
+    delete logEntry.timestamp;
+    assert.deepEqual(logEntry, {
+      level: 'info',
+      message: 'Hello, world',
+      myId: 42,
+      ow: {
+        actionName: 'test-my-action-name',
+        activationId: 'test-my-activation-id',
+        transactionId: 'test-transaction-id',
+      },
+    });
+  });
+
+  it('openhwisk creates papertrail logger if needed', async () => {
+    const reqs = [];
+    let server;
+    const srv = new Promise((resolve, reject) => {
+      server = net.createServer((socket) => {
+        socket.on('data', (data) => {
+          reqs.push(...data.toString().trim().split('\n'));
+          socket.end();
+          server.close();
+        }).on('close', resolve);
+      })
+        .on('error', (err) => {
+          reject(err);
+        });
+    });
+
+    const port = await new Promise((resolve) => {
+      server.listen(() => {
+        resolve(server.address().port);
+      });
+    });
+
+    const log = logger.init({
+      PAPERTRAIL_HOST: '127.0.0.1',
+      PAPERTRAIL_PORT: port,
+      PAPERTRAIL_TLS: 'false',
+      PAPERTRAIL_LOG_LEVEL: 'info',
+    }, myRootLogger);
+
+    log.info({ myId: 42 }, 'Hello, world');
+
+    await srv;
+
+    assert.equal(reqs.length, 2);
+    assert.ok(reqs[0].endsWith('test-my-action-name[1]:test-my-activati INFO Hello, world'));
```

Comment:
Why `test-my-activati` is trimmed here ?

Reply:
I can't remember exactly. but I thought that papertrail had a limit on the size of the `procid` field of the message.